apps/homescreen/js/pages.js

```diff
@@ -219,14 +219,11 @@
       // Remove child immediately, datastore operations can be quite slow
       this.removeCard(this.selectedCard);
 
-      this.pagesStore.get(id).then(entry => {
-        entry.data.pinned = false;
-        this.pagesStore.datastore.put(entry.data, id).then(() => {},
-          e => {
-            console.error('Error unpinning page:', e);
-          });
-      }, e => {
-        console.error('Error retrieving page to unpin:', e);
+      // TODO: fix LazyLoader mock to support promises
+      return new Promise((resolve, reject) => {
+        LazyLoader.load(['shared/js/places_model.js'], () => {
```

Comment:
This is a lot of code to do this small check, I'd rather a utility function was added to apps/homescreen/js/datastore.js to put values in the datastore and do the revision check there.